### DISCUSSION TITLE
Code sample embeds in docs site should move away from CodeSandbox

### DISCUSSION THREAD

**Discussion by thanhtran-ct** [pmndrs/jotai#2516] [Ideas] [answered]
As a Jotai newbie, it's quite annoyed that these code examples are presented with out of credit warning from codesandbox.
![Screenshot 2024-04-23 at 8 50 53 AM](https://github.com/pmndrs/jotai/assets/99478251/cccd5649-40c7-4e63-bca8-f431b508c955)
Even though I can fork the examples to my own codesandbox account, it's quite tedious and eliminates the purpose of code embeds in the docs. Besides, it's also troublesome for contributors to help updating the code examples when there are updates to the library.

**Comment by dai-shi** [comment]
Yes, I totally agree with you.
Would you mind working on it?
Some options are StackBlitz and LiveCodes.

**Reply by thanhtran-ct** [reply]
I'm eager to help. But I don't have much experience to recommend which service suitable to use. Please let me know which one you'd choose.

**Reply by dai-shi** [reply, marked as answer]
I just realized we already support `<Stackblitz>`.

https://github.com/pmndrs/jotai/blob/5a525d82b5cfa68529f09f92701519ce13deef37/docs/guides/vite.mdx?plain=1#L30-L34

Let's use it. Please open a small PR first to change one or two for the start.

**Reply by mayank99** [reply]
fwiw, stackblitz embeds don't work in non-chromium browsers. https://developer.stackblitz.com/platform/webcontainers/browser-support#embedding

<details><summary>Screenshots</summary>

![](https://github.com/pmndrs/jotai/assets/9084735/a58abc9a-d2b5-4cb0-a940-753736907678)
![](https://github.com/pmndrs/jotai/assets/9084735/3cad64db-b514-4645-b4a2-205b716734f4)

</details>

**Reply by dai-shi** [reply]
Should we put link separately, or the link in the embed one would be fine?

**Reply by mayank99** [reply]
I actually couldn't get the "Fork" link from the embed to work.

<img src="https://github.com/pmndrs/jotai/assets/9084735/1f1830cd-63af-4f3f-8e4b-ebfc5228867f" width="400">

Even if it did work though, it is a worse experience than having the embed working correctly in the page.

If the main issue with Codesandbox is the pricing limitation, it might be worth reaching out to them. They have an [OSS program](https://codesandbox.io/docs/learn/plans/codesandbox-friends), and they might even provide free access to Pro plans to prominent OSS project maintainers (for use across multiple projects).

**Reply by dai-shi** [reply]
@mayank99 I wonder if you prefer LiveCodes. It doesn't work for server-side, but most of our examples are client only.

**Reply by mayank99** [reply]
Hadn't heard of LiveCodes before, but I looked into it just now and I think it might be the worst option of the bunch. The whole experience is very confusing, with important details obfuscated (e.g. which package versions are used, how is bundler configured, etc), and the UI is built in a shoddy and inaccessible way (e.g. divs as buttons, random `tabindex` values, etc).

If you're ok with fully client-side though, maybe [CodePen](https://codepen.io/features/embeds) could be an option too? It has opt-in support for JSX and TypeScript.

Another good option is [Sandpack](https://sandpack.codesandbox.io/), which I think doesn't suffer from the same pricing limitations as regular Codesandbox.

**Reply by dai-shi** [reply]
Thanks for the suggestion. Okay, let's stick with StackBlitz then. I guess it has the best experience for now.

Can anyone make https://github.com/pmndrs/jotai/blob/02cf0baf27bed79d0b284e3e2ef23a5f00e6908c/website/src/components/stackblitz.js so that it only uses embedding if it's the supported browser? Otherwise, just show the link.
